backend/penndata/views.py

```diff
@@ -268,6 +270,125 @@ def get_queryset(self):
         return recent_snapshots
 
 
+class FitnessUsage(APIView):
+    def safe_add(self, a, b):
+        if a is None:
+            return b
+        if b is None:
+            return a
+        return a + b
+
+    def get_usage_on_date(self, room, date, field):
+        """
+        Returns the number of people in the fitness center on a given date per hour
+        """
+
+        # get all snapshots for a given room on a given date and the days before and after
+        snapshots = FitnessSnapshot.objects.filter(
+            room=room,
+            date__gte=date - datetime.timedelta(days=1),
+            date__lte=date + datetime.timedelta(days=1),
+        )
+
+        usage = [None] * 24  # None represents no data so initialize to all None
+        for hour in range(24):
+            try:
+                # consider the :30 mark of each hour
+                hour_date = timezone.make_aware(
+                    datetime.datetime.combine(date, datetime.time(hour)), is_dst=None
+                ) + datetime.timedelta(minutes=30)
+            except NonExistentTimeError:
```

Comment:
i was testing this on day light savings day and 2am didn't exist so i was trying things. don't think i need it anymore since i have the exception... will remove